### PR TITLE
[Inference API] Deprecate elser service

### PR DESCRIPTION
**Pull request by maxhniebergall**
This change adds a deprecation warning for the elser service on `Put` and `Infer` and copies the elser service files so we can simply delete the elser service after 8.16

This change maintains BWC with the elser service.

### CODE REVIEW

**Review by davidkyle — COMMENTED** [review]
Now that the elasticsearch service can run the elser model can we remove the elser service in this PR? The elasticsearch service can be an alias to the elser service, adding some logic to the `InferenceServiceRegistry` lookup code would simply return the elasticsearch service when elser is requested.

**Comment by maxhniebergall** [comment]
@elasticmachine merge upstream

**Review by davidkyle — APPROVED** [review]
LGTM

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportInferenceAction.java [hunk 1]
```diff
@@ -75,6 +80,15 @@ protected void doExecute(Task task, InferenceAction.Request request, ActionListe
...
             }
+            if (service.get().name().equals(OLD_ELSER_SERVICE_NAME)) {
+                DEPRECATION_LOGGER.warn(
+                    DeprecationCategory.API,
+                    "inference_api_elser_service",
+                    "The [{}] service is deprecated and will be removed in a future release. Use the [{}] service instead.",
+                    OLD_ELSER_SERVICE_NAME,
+                    ElasticsearchInternalService.NAME
+                );
+            }
```
Comment (davidkyle) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportInferenceAction.java:91]:
```suggestion
```

**Pull request merged**